federation/pkg/federatedtypes/hpa.go

```diff
@@ -165,7 +169,7 @@ func (a *HpaAdapter) NewTestObject(namespace string) pkgruntime.Object {
 		},
 		Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
 			ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
-				Kind: "replicaset",
+				Kind: "ReplicaSet",
```

Comment:
This is actually used only for crud test written with the sync controller (which tests crud of the hpa object alone). Meanwhile, I will add tests for both types in e2e.